src/Raven.Studio/typescript/components/pages/database/documents/identities/DocumentIdentitiesModal.tsx

```diff
@@ -0,0 +1,182 @@
+        context: {
+            identities,
+            isEditing,
+        },
+        resolver: yupResolver(documentIdentitiesSchema),
+        defaultValues,
+    });
+
+    const formValues = useWatch({ control: form.control });
+
+    const onSubmit: SubmitHandler<AddIdentitiesFormData> = ({ prefix, value }) => {
+        return tryHandleSubmit(async () => {
+            await databasesService.seedIdentity(databaseName, prefix, value);
+            form.reset();
+            refetch();
+            toggleModal();
+        });
+    };
+
+    return (
+        <Modal contentClassName="modal-border bulge-primary" wrapClassName="bs5" size="lg" {...props}>
+            <form onSubmit={form.handleSubmit(onSubmit)}>
+                <ModalBody>
+                    <div className="position-absolute m-2 end-0 top-0">
+                        <Button close onClick={toggleModal} />
+                    </div>
+                    <div className="w-100 d-flex align-items-center justify-content-center flex-column">
+                        <Icon size="xl" icon="identities" color="primary" margin="me-0" />
+                        <h4>{isEditing ? "Edit Identity" : "Add new identity"}</h4>
+                    </div>
+                    <div className="w-100 d-flex flex-column gap-4 mb-4">
+                        <DocumentIdentitiesModalForm isEditing={isEditing} control={form.control} />
+                    </div>
+                    <InformationBadge isEditing={isEditing} {...formValues} />
+                </ModalBody>
+                <ModalFooter>
+                    <Button className="link-muted" color="link" onClick={toggleModal} type="button">
+                        Close
+                    </Button>
+                    <ButtonWithSpinner
+                        className="rounded-pill"
+                        color="success"
+                        icon="save"
+                        isSpinning={form.formState.isSubmitting}
+                        type="submit"
+                    >
+                        Save identity
+                    </ButtonWithSpinner>
+                </ModalFooter>
+            </form>
+        </Modal>
+    );
+}
+
+interface InformationBadgeProps extends AddIdentitiesFormData {
+    isEditing: boolean;
+}
+
+function InformationBadge({ prefix = "<Prefix>", value, isEditing }: InformationBadgeProps) {
+    const databaseName = useAppSelector(databaseSelectors.activeDatabaseName);
+    const { manageServerService } = useServices();
+    const { result: identityPartsSeparator, loading } = useAsync(async () => {
+        const clientConfiguration = manageServerService.getClientConfiguration(databaseName);
+        const globalClientConfiguration = manageServerService.getGlobalClientConfiguration();
+
+        const [clientConfig, globalConfig] = await Promise.allSettled([clientConfiguration, globalClientConfiguration]);
```

Comment:
I think we can use Promise.all here, as we handle 404 internally in getClient/Global configuration. 
So we either get proper value or null. 

In case of error it will simply throw.

Reply:
This was my idea with allSettled, but you are right, it's actually better to handle something like that directly in command